pkg/billing/client.go

```diff
@@ -5,34 +5,59 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"math/rand"
 	"net/http"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
 )
 
 type Client struct {
-	BaseURL string
-	httpc   *http.Client
+	BaseURL  string
+	httpc    *http.Client
+	hostname string
 }
 
-func NewClient(url string, c *http.Client) Client { return Client{BaseURL: url, httpc: c} }
+func NewClient(url string, c *http.Client) Client {
+	hostname, err := os.Hostname()
+	if err != nil {
+		hostname = fmt.Sprintf("unknown-%d", rand.Intn(1000))
+	}
+	return Client{BaseURL: url, httpc: c, hostname: hostname}
+}
 
 func (c Client) NewBatch() *Batch { return &Batch{c: c, events: nil} }
 
 type Batch struct {
 	// Q: does this need a mutex?
 	c      Client
-	events []json.Marshaler
+	events []any
 }
 
 // Count returns the number of events in the batch
 func (b *Batch) Count() int {
 	return len(b.events)
 }
 
+func (b *Batch) idempotenize(key string) string {
+	if key != "" {
+		return key
+	}
+
+	return fmt.Sprintf("Host<%s>:ID<%s>:T<%s>", b.c.hostname, uuid.NewString(), time.Now().Format(time.RFC3339))
```

Comment:
NIT: having a space as a separator for RFC3339 time is slightly easier to parse.

Reply:
are we going to be parsing these or just reading them?

Reply:
We're not going to parse them.